code/game/objects/items/devices/scanners.dm

```diff
@@ -53,42 +55,31 @@ REAGENT SCANNER
 	scan()
 
 /obj/item/t_scanner/proc/scan()
+	t_ray_scan(src.loc, pulse_duration=pulse_duration, scan_range=scan_range)
 
-	for(var/turf/T in range(scan_range, src.loc) )
-
-		if(!T.intact)
-			continue
+// Show the viewer the T-ray layout within `scan_range` for the `pulse_duration` deciseconds.
+/proc/t_ray_scan(mob/viewer, pulse_duration, scan_range)

```

Comment:
Why is this its own proc instead of just doing this code inside `proc/scan()`?

Reply:
Because that's the way it was done on tg.
They have both the hand held scanner and T-ray goggles, which both call this proc.

Honestly, I'm still very unsure of how close I should keep to the tg original when doing ports, especially when it comes to small things like this. I might be reading too much into this section of the contributing guide here:

> Document and explain your pull requests thoroughly. Failure to do so will delay a PR as we question why changes were made. This is especially important if you're porting a PR from another codebase (i.e. TG) and *divert from the original*.

(emphasis mine) but that was the motivation for keeping stuff as close to the port source as possible.

Should i remove the standalone proc?

Reply:
Oh, this will explain why the proc isn't typecast.

What I would do is make it a t-ray scanner level proc for now and then repath it once you port the goggles. I think the maintainers would be ok with it being a different path to tg code (obviously I'm not speaking for them here though)

Reply:
Removed standalone proc. Thanks!